### PR TITLE
ci: Change the way to launch Docker containers

### PR DESCRIPTION
**Pull request by kuisathaverat**
fix: define GOPATH to allow use go modules cache
fix: grab junit results
fix: quiet Docker build
chore: split protobuf go modules install

### CODE REVIEW

**Comment by kuisathaverat** [comment]
🤔 why the CI said that does not found `docker`

**Comment by kuisathaverat** [comment]
well, the `docker` command is running in the master, and it does not have Docker installed so we have to change the approach. So we can not use this on beats-ci or apm-ci, locally works because we have Docker installed locally.

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -103,3 +82,15 @@ pipeline {
     }
   }
 }
+
+def insideGo(Closure body){
+  def goAgent = docker.build("go-agent", "${BASE_DIR}/.ci/jenkins-go-agent")
```
Comment (v1v) [inline .ci/Jenkinsfile:87]: retry for the win!

```suggestion
  def goAgent
  retry(2) {
    sleep randomNumber(min: 2, max: 5)
    goAgent = docker.build("go-agent", "${BASE_DIR}/.ci/jenkins-go-agent")
  }
```

**Comment by kuisathaverat** [comment]
I have seen this stack trace a few times, also locally, I am not sure what it is related to

```
Error
Failed
Stacktrace
go: downloading google.golang.org/protobuf v1.23.0
go: downloading google.golang.org/grpc v1.29.1
go: downloading github.com/stretchr/testify v1.5.1
go: downloading github.com/golang/protobuf v1.4.2
go: downloading github.com/gofrs/uuid v3.3.0+incompatible
go: downloading github.com/davecgh/go-spew v1.1.0
go: downloading gopkg.in/yaml.v2 v2.2.2
go: downloading github.com/pmezard/go-difflib v1.0.0
go: downloading google.golang.org/genproto v0.0.0-20190819201941-24fa4b261c55
go: downloading golang.org/x/net v0.0.0-20190311183353-d8887717615a
go: downloading golang.org/x/sys v0.0.0-20190215142949-d0b11bdaac8a
go: downloading golang.org/x/text v0.3.0
runtime: bad pointer in frame google.golang.org/grpc.(*Server).Serve at 0xc000051ed0: 0xf
fatal error: invalid pointer found on stack
runtime stack:
runtime.throw(0xa29acc, 0x1e)
runtime.adjustpointers(0xc000051eb8, 0x7ffd6300e228, 0x7ffd6300e5b8, 0xe3b9e8, 0xe7f260)
runtime.adjustframe(0x7ffd6300e4c8, 0x7ffd6300e5b8, 0xe7f260)
runtime.gentraceback(0xffffffffffffffff, 0xffffffffffffffff, 0x0, 0xc000106c00, 0x0, 0x0, 0x7fffffff, 0xa43138, 0x7ffd6300e5b8, 0x0, ...)
runtime.copystack(0xc000106c00, 0x1000)
runtime.newstack()
runtime.morestack()
goroutine 20 [copystack]:
runtime.mallocgc(0x40, 0x0, 0x0, 0x599c55)
runtime.rawstring(0x40, 0x0, 0x0, 0x0, 0x0, 0x0)
runtime.rawstringtmp(0x0, 0x40, 0x0, 0xc000040458, 0x41682f, 0x7f82ec153700, 0x20300000000000)
runtime.concatstrings(0x0, 0xc000051cd8, 0x2, 0x2, 0xae34e0, 0x400)
runtime.concatstring2(0x0, 0xa211df, 0xf, 0xa338af, 0x31, 0xa5af01, 0xae34e0)
runtime.errorString.Error(...)
runtime.(*errorString).Error(0xeb6bc0, 0x974dc0, 0xeb6bc0)
runtime.preprintpanics(0xc000051de8)
panic(0x974dc0, 0xeb6bc0)
runtime.panicmem(...)
runtime.sigpanic()
google.golang.org/grpc.(*Server).Serve(0x0, 0xaeb6c0, 0xc00012b2c0, 0x0, 0x0)
github.com/elastic/elastic-agent-client/v7/pkg/client.(*StubServer).Start.func1(0xc0001cb320, 0xaeb6c0, 0xc00012b2c0)
runtime.goexit()
created by github.com/elastic/elastic-agent-client/v7/pkg/client.(*StubServer).Start
goroutine 1 [chan receive]:
runtime.gopark(0xa431a0, 0xc000102838, 0x170e, 0x2)
runtime.chanrecv(0xc0001027e0, 0xc0001fdc87, 0xc000000101, 0x5026d7)
runtime.chanrecv1(0xc0001027e0, 0xc0001fdc87)
testing.(*T).Run(0xc000222000, 0xa28c40, 0x1d, 0xa41a58, 0x48ab01)
testing.runTests.func1(0xc000161c20)
testing.tRunner(0xc000161c20, 0xc0001fde10)
testing.runTests(0xc00012b2a0, 0xec0500, 0x8, 0x8, 0x0)
testing.(*M).Run(0xc0001c0380, 0x0)
main.main()
runtime.main()
runtime.goexit()
goroutine 2 [force gc (idle)]:
runtime.gopark(0xa43390, 0xecabd0, 0x1411, 0x1)
runtime.goparkunlock(...)
runtime.forcegchelper()
runtime.goexit()
created by runtime.init.6
goroutine 3 [GC sweep wait]:
runtime.gopark(0xa43390, 0xecae40, 0x140c, 0x1)
runtime.goparkunlock(...)
runtime.bgsweep(0xc000030150)
runtime.goexit()
created by runtime.gcenable
goroutine 4 [GC scavenge wait]:
runtime.gopark(0xa43390, 0xecae00, 0x140d, 0x1)
runtime.goparkunlock(...)
runtime.bgscavenge(0xc000030150)
runtime.goexit()
created by runtime.gcenable
goroutine 18 [finalizer wait]:
runtime.gopark(0xa43390, 0xef62e8, 0x101410, 0x1)
runtime.goparkunlock(...)
runtime.runfinq()
runtime.goexit()
created by runtime.createfing
goroutine 21 [sleep]:
runtime.gopark(0xa433c8, 0xc000100d70, 0x481313, 0x1)
time.Sleep(0x989680)
github.com/elastic/elastic-agent-client/v7/pkg/client.waitFor(0xc00015bed8, 0xc000222000, 0x0)
github.com/elastic/elastic-agent-client/v7/pkg/client.TestClient_Checkin_With_Token(0xc000222000)
testing.tRunner(0xc000222000, 0xa41a58)
runtime.goexit()
created by testing.(*T).Run
goroutine 22 [runnable]:
github.com/elastic/elastic-agent-client/v7/pkg/client.(*StubServer).Start.func1(0xc0001cb5f0, 0xaeb6c0, 0xc00012b3a0)
runtime.goexit()
created by github.com/elastic/elastic-agent-client/v7/pkg/client.(*StubServer).Start
goroutine 23 [runnable]:
google.golang.org/grpc.(*ccBalancerWrapper).watcher(0xc000155500)
runtime.goexit()
created by google.golang.org/grpc.newCCBalancerWrapper
goroutine 24 [runnable]:
google.golang.org/grpc.(*addrConn).resetTransport(0xc000224000)
runtime.goexit()
created by google.golang.org/grpc.(*addrConn).connect
goroutine 25 [runnable]:
github.com/elastic/elastic-agent-client/v7/pkg/client.(*Client).startCheckin.func1(0xc0002200f0)
runtime.goexit()
created by github.com/elastic/elastic-agent-client/v7/pkg/client.(*Client).startCheckin
goroutine 26 [select]:
runtime.gopark(0xa433e0, 0x0, 0x1809, 0x1)
runtime.selectgo(0xc000050a50, 0xc0000509c8, 0x2, 0x0, 0x0)
google.golang.org/grpc.(*pickerWrapper).pick(0xc0001cb890, 0xaecb40, 0xc0001cbad0, 0x40df01, 0xa278b5, 0x1b, 0xaecb40, 0xc0001cbad0, 0xc000050b18, 0x4f827c, ...)
google.golang.org/grpc.(*ClientConn).getTransport(0xc00020ea80, 0xaecb40, 0xc0001cbad0, 0xc000222201, 0xa278b5, 0x1b, 0xecc4e0, 0x7f8312c74e98, 0x0, 0x30, ...)
google.golang.org/grpc.(*clientStream).newAttemptLocked(0xc000222240, 0x0, 0x0, 0x0, 0x0, 0x0)
google.golang.org/grpc.newClientStream(0xaecb40, 0xc0001cbad0, 0xebf1c0, 0xc00020ea80, 0xa278b5, 0x1b, 0x0, 0x0, 0x0, 0x0, ...)
google.golang.org/grpc.(*ClientConn).NewStream(0xc00020ea80, 0xaeca80, 0xc000155580, 0xebf1c0, 0xa278b5, 0x1b, 0x0, 0x0, 0x0, 0x4f612b, ...)
github.com/elastic/elastic-agent-client/v7/pkg/proto.(*elasticAgentClient).Actions(0xc0001fe5b0, 0xaeca80, 0xc000155580, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
github.com/elastic/elastic-agent-client/v7/pkg/client.(*Client).startActions.func1(0xc0002200f0, 0xc000120ea0)
runtime.goexit()
created by github.com/elastic/elastic-agent-client/v7/pkg/client.(*Client).startActions
```

**Comment by kuisathaverat** [comment]
sorry I forget to change the button to squash+merge 🤦

**Comment by ph** [comment]
no worries.. I've made the change to only squash+merge :)

**Pull request merged**